lib/pbench/agent/tool_data_sink.py

```diff
@@ -739,28 +750,23 @@ class ToolDataSink(Bottle):
     @staticmethod
     def fetch_params(params, pbench_run):
         try:
-            _benchmark_run_dir = params["benchmark_run_dir"]
-            bind_hostname = params["bind_hostname"]
-            port = params["port"]
-            channel_prefix = params["channel_prefix"]
-            tool_group = params["group"]
-            tool_metadata = ToolMetadata.tool_md_from_dict(params["tool_metadata"])
-            tool_trigger = params["tool_trigger"]
-            tools = params["tools"]
+            _benchmark_run_dir = BenchmarkRunDir(
+                params["benchmark_run_dir"], pbench_run
+            )
+            tdsp = ToolDataSinkParams(
+                benchmark_run_dir=_benchmark_run_dir,
+                bind_hostname=params["bind_hostname"],
+                port=params["port"],
+                channel_prefix=params["channel_prefix"],
+                tool_group=params["group"],
+                tool_metadata=ToolMetadata.tool_md_from_dict(params["tool_metadata"]),
+                tool_trigger=params["tool_trigger"],
+                tools=params["tools"],
+            )
```

Comment:
I would get rid of the local variables:
```
            return ToolDataSinkParams(
                benchmark_run_dir=BenchmarkRunDir(
                    params["benchmark_run_dir"], pbench_run
                ),
                bind_hostname=params["bind_hostname"],
                port=params["port"],
                channel_prefix=params["channel_prefix"],
                tool_group=params["group"],
                tool_metadata=ToolMetadata.tool_md_from_dict(params["tool_metadata"]),
                tool_trigger=params["tool_trigger"],
                tools=params["tools"],
            )

```

Reply:
So this change came about by breaking a larger commit that was going to land in PR #2743 into smaller ones.  What is coming is a [change](https://github.com/distributed-system-analysis/pbench/pull/2743/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-f186c241cb95e4c5972607dfea8d6669f4cb48c20a0def4276f58ad08808214dR771) to use that variable.